zulipterminal/ui_tools/boxes.py

```diff
@@ -162,8 +166,13 @@ def _stream_box_autocomplete(self, text: str, state: Optional[int]
                                        self.view.pinned_streams)
 
         # matched_streams[0] and matched_streams[1] contains the same data.
-        return self._process_typeaheads(matched_streams[0], state,
+        processed_typeaheads = self._process_typeaheads(matched_streams[0], state,
                                         matched_streams[1])
+        if(processed_typeaheads):
+            color = streams_list[streams.index(processed_typeaheads)]['color']
+            self.header_write_box[0].set_caption((color, '█ '))
+            self.header_write_box[1].set_caption((color, STREAM_TOPIC_SEPARATOR+' '))
```

Comment:
By setup we mean the` __init__()` function right?

Reply:
Yes - we set stream and topic captions (with the color) at different times, so it would be nice to centralize it. That said, the code is slightly different.